### PR TITLE
Code improvements & bug fixes for OP-GAN

### LINKED ISSUE #11: using cpu
how do we use cpu for sampling or at least validating

### PR DESCRIPTION
**Pull request by rubencart**
There were some bugs that caused the code to crash (both with dependencies in original requirements.txt and with newer package versions from environment.yml). I fixed the ones I encountered (there may be more).

Also, I tried to improve (imo) some aspects of the code. For instance, model parameters are now much more configurable.

Changes:
- conda instead of pip
- remove need to do `cd code`, just use `OP-GAN` as working dir, which is nicer because than you can specify data and output dirs etc as `data/` instead of `../data/`
- prints learnable nb of params per network
- fixed intermediate image exports (build_super_images), there were bugs with the batch_size being a list, a hard coded number 8 that gives an error (maybe only when current batch_size is smaller than 8), a parameter to a scikit-image function to indicate there is a channel dimension.
- moved a lot of constants and hard-to-interpret dimension names from model.py to configs, so they are easier to understand and more configurable
- wraps networks in `torch.nn.DataParallel` when n_gpu > 1, so `torch.nn.data_parallel` doesn't have to be repeated everywhere
- uses `.to(device)` instead of `.cuda()` for more device-agnostic code (works on CPU as well)
- uses the `CUDA_VISIBLE_DEVICES` environment variable, so pytorch can only see the GPUs you want it to use, so it won't appear as using unspecified GPUs with 0MB in nvidia-smi
- allows to actually set the random seed manually, in config .yml (wasn't the case before)
- fixes image resizing in preprocessing so you can use less than 3 discriminators if you want to (gave an error before)
- adds `DEBUG` and `DEBUG_NUM_DATAPOINTS` parameters to config, so to debug, you can use a smaller version of the dataset
- changes all print statements to proper logging, with a timestamp etc, automatically saves these logs to files
- fixes a number of deprecation warnings that were displayed, like a deprecated overload of torch's `add_`

One of the warnings was the following: `UserWarning: Default grid_sample and affine_grid behavior has changed to align_corners=False since 1.3.0. Please specify align_corners=True if the old behavior is desired. See the documentation of grid_sample for details` [docs](https://pytorch.org/docs/stable/nn.functional.html#grid-sample). I set `align_corners=False` in `stn` in model.py, because this seems better, but since the code was developed with pytorch 1.1, and the default behavior was only changed in pytorch 1.3, `align_corners=True` is likely to be what you used (it was the default). Do you think we should keep it at `align_corners=True`?

### CODE REVIEW

**Comment by tohinz** [comment]
Thanks a lot for this, appreciated.
